cmd/migrate/v3/migrate.go

```diff
@@ -0,0 +1,122 @@
+package v3
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	parserconfig "github.com/forbole/juno/v3/parser/config"
+
+	"gopkg.in/yaml.v3"
+
+	v2 "github.com/forbole/juno/v3/cmd/migrate/v2"
+	"github.com/forbole/juno/v3/cmd/parse"
+	"github.com/forbole/juno/v3/database"
+	databaseconfig "github.com/forbole/juno/v3/database/config"
+	v3db "github.com/forbole/juno/v3/database/legacy/v3"
+	"github.com/forbole/juno/v3/database/postgresql"
+	"github.com/forbole/juno/v3/types/config"
+)
+
+// RunMigration runs the migrations from v2 to v3
+func RunMigration(parseConfig *parse.Config) error {
+	// Migrate the config
+	cfg, err := migrateConfig()
+	if err != nil {
+		return fmt.Errorf("error while migrating config: %s", err)
+	}
+
+	// Refresh the global configuration
+	err = parse.UpdatedGlobalCfg(parseConfig)
+	if err != nil {
+		return err
+	}
+
+	bz, err := yaml.Marshal(&cfg)
+	if err != nil {
+		return fmt.Errorf("error while serializing config: %s", err)
+	}
+
+	err = ioutil.WriteFile(config.GetConfigFilePath(), bz, 0666)
+	if err != nil {
+		return fmt.Errorf("error while writing v3 config: %s", err)
+	}
+
+	// Migrate the database
+	err = migrateDb(cfg, parseConfig)
+	if err != nil {
+		return fmt.Errorf("error while migrating database: %s", err)
+	}
+
+	return nil
+}
+
+func migrateConfig() (Config, error) {
+	cfg, err := v2.GetConfig()
+	if err != nil {
+		return Config{}, fmt.Errorf("error while reading v2 config: %s", err)
+	}
+
+	// Get the new fields added inside the various configurations
+	var partitionSize int64
+	if cfg.Database.PartitionSize != nil {
+		partitionSize = *cfg.Database.PartitionSize
+	}
+
+	var partitionBatchSize int64
+	if cfg.Database.PartitionBatchSize != nil {
+		partitionBatchSize = *cfg.Database.PartitionBatchSize
+	}
+
+	var averageBlockTime = 3 * time.Second
```

Comment:
I've set the average block time to 3 seconds because there are some chains (eg. e-Money) that have a slower block time. Also, it's safer to have a slightly faster query period rather than a slower one (which might lead to some blocks being parsed too late). If you think it's best I can set the default to `5` seconds though